test/test_hsl_ma57_patch.jl

```diff
@@ -26,6 +26,8 @@ function test_ma57_patch(A, M, b, xexact)
   d1 = abs.(diag(D))
   d2 = [diag(D, 1); 0][:]
   ma57_alter_d(M, [Vector(d1)'; Vector(d2)'])
+  x4 = ma57_solve(M, b)
+  @test norm(x4 - xexact) ≤ ϵ * norm(xexact)
```

Comment:
This couldn't possibly hold since you've modified the factors.

Reply:
Oups, I updated the test.
I checked the Julia code and it's a copy of the Fortran code.
Tjey probably never tested this routine or only with matrices with 1x1 or 2x2 blocks.

Reply:
If you change the factors, you no longer have a factorization of A. So there's no reason to expect that x4 would be close to xexact. It's the solution to a different system.

Reply:
Ah sorry, I did not see the updated test. Yes, that's a problem.

Reply:
If you extract D again, what's inside?

Reply:
I found the problem, `ma57_inter_d` expect that you provide `D^{-1}` and not `D`...

```julia
using HSL

T = Float64
rows = Cint[1, 1, 2, 2, 3, 3, 5]
cols = Cint[1, 2, 3, 5, 3, 4, 5]
vals = T[2, 3, 4, 6, 1, 5, 1]
A = sparse(rows, cols, vals)
A = A + triu(A, 1)'

M = ma57_coord(5, rows, cols, vals)

b = T[8, 45, 31, 15, 17]
xexact = T[1, 2, 3, 4, 5]

ϵ = sqrt(eps(eltype(A)))
ma57_factorize!(M)

# extract factors
(L, D, s, p) = ma57_get_factors(M)

invD = inv(Matrix(D))
d1 = diag(invD)
d2 = [diag(invD, 1); 0][:]
Dnew = [Vector(d1)'; Vector(d2)']

ma57_alter_d(M, Dnew)

(L2, D2, s2, p2) = ma57_get_factors(M)

@test norm(D - D2) ≤ ϵ

x4 = ma57_solve(M, b)
@test norm(x4 - xexact) ≤ ϵ * norm(xexact)
```

Reply:
I update the docstrings.
Dominique, they added your patch in the release 5.3.2 for information.

Reply:
Phew! Yes, I thought my patch made it into the latest release. As a backup, I mentioned to @AntoninKns that the functionality also comes standard in MA97 (though I forgot to document it).